sdk/ml/azure-ai-ml/azure/ai/ml/entities/_component/component.py

```diff
@@ -449,23 +452,30 @@ def __call__(self, *args, **kwargs) -> [..., Union["Command", "Parallel"]]:
     @contextmanager
     def _resolve_local_code(self):
         """Resolve working directory path for the component."""
-        if hasattr(self, "code"):
-            code = getattr(self, "code")
-            # Hack: when code not specified, we generated a file which contains
-            # COMPONENT_PLACEHOLDER as code
-            # This hack was introduced because job does not allow running component without a
-            # code, and we need to make sure when component updated some field(eg: description),
-            # the code remains the same. Benefit of using a constant code for all components
-            # without code is this will generate same code for anonymous components which
-            # enables component reuse
-            if code is None:
-                with tempfile.TemporaryDirectory() as tmp_dir:
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            if hasattr(self, "code"):
+                code = getattr(self, "code")
+                # Hack: when code not specified, we generated a file which contains
+                # COMPONENT_PLACEHOLDER as code
+                # This hack was introduced because job does not allow running component without a
+                # code, and we need to make sure when component updated some field(eg: description),
+                # the code remains the same. Benefit of using a constant code for all components
+                # without code is this will generate same code for anonymous components which
+                # enables component reuse
+                if code is None:
                     code = Path(tmp_dir) / COMPONENT_PLACEHOLDER
                     with open(code, "w") as f:
                         f.write(COMPONENT_CODE_PLACEHOLDER)
                     yield code
+                # if component's code is a single file, we can directly use it without change;
+                # otherwise we need to copy to temporary folder filtering __pycache__.
+                else:
+                    if os.path.isfile(code):
+                        yield code
+                    else:
+                        src_path = Path(self.base_path) / code
+                        dst_path = Path(tmp_dir) / TEMP_COMPONENT_CODE_FOLDER / src_path.name
```

Comment:
what is the TEMP_COMPONENT_CODE_FOLDER for?

Reply:
it's a mark for code asset in temp folder, maybe we can remove this with better mock function for upload code asset

Reply:
As offline discussed, please create a separate task to track the exact reason that we need to do this. It looks wired to put such a constant here.

A temporary solution can be update sanitizer in conftest if you're sure that there will only be diff in request uri

Reply:
I think I have misunderstood this component, this constant mark shall be removed by only changing some code in test.
let's wait for the new CI result